feat(date-picker): support type="week"

suica opened the pull request:
fixes #4525 
<!--
!!! Please read the following content if this is your first pull request !!!

1. If you are working on docs, please set `docs` branch as the target branch.
2. If you are working on bug fixes or new features, please set `main` branch as the target branch.

For people who are working on 2, please add changelog to `CHANGELOG.zh-CN.md` & `CHANGELOG.en-US.md` in the PR. You need to add changelog to both files. You can use English in both file. The develop team will translate it later.

About docs & changelog's format and other tips, see in `CONTRIBUTING.md`.
-->
<!--
!!! 如果这是你第一次提交 PR，请阅读下面的内容 !!!

1. 如果你在修改文档，请提交到 `docs` 分支
2. 如果你在修复 Bug 或者开发新的特性，请提交到 `main` 分支

对于进行工作 2 的人，请在 `CHANGELOG.zh-CN.md` & `CHANGELOG.en-US.md` 中添加变更日志，你需要在两个文件中都添加变更日志。你可以只使用中文，开发团队会在之后进行翻译。

关于文档和变更日志的格式以及其他的帮助信息，请参考 `CONTRIBUTING.md`。
-->

## Todos
- [x] finish `type="week"`
- [ ] check disabled style
- [x] check the i18n files for adding new localized text
- [ ] check the enUS doc
- [x] update the changelog
- [ ] clean the code

suica commented on src/date-picker/src/DatePicker.tsx:
```diff
@@ -387,6 +391,10 @@ export default defineComponent({
           return result
         }
+        case 'week': {
+          result.push('confirm')
```
Maybe we also don't need this

suica replied:
Removed. Now, `week` and `date` share the same case branch.

Sepush commented:
demos for both en-US and zh-CN but you can add it finally

Sepush commented:
when impl range need disable the date before the start time

Sepush commented:
in week mode we do not need the hover style for the speific day 
![image](https://github.com/tusen-ai/naive-ui/assets/39197136/cb7b08f2-6aff-4673-bb0d-027d64e7c7e4)
the value/the arg for `on-update:value` the day of firt day of week not the click 
also need handle the `default-time` prop

suica commented:
Since type `date` and `week` share a simlilar outlook, most code could be resused. So i chose to extend `Date.tsx` to make it support an extra `week` mode.
Do you think it is a good idea? Should we copy the code and make a new week panel?

suica commented:
> demos for both en-US and zh-CN but you can add it finally

Added.



> in week mode we do not need the hover style for the speific day ![image](https://user-images.githubusercontent.com/39197136/253610389-cb7b08f2-6aff-4673-bb0d-027d64e7c7e4.png) the value/the arg for `on-update:value` the day of firt day of week not the clicked day also need handle the `default-time` prop

Fixed. Removed hover effect for both `--slightly-covered` and `--heavily-covered` modifiers.

@Sepush And, do you think we need a new dedicated file for `week` panel? Now extending the original date panel seems to be a little bit awkward...

Sepush commented:
> Since type `date` and `week` share a simlilar outlook, most code could be resused. So i chose to extend `Date.tsx` to make it support an extra `week` mode. Do you think it is a good idea? Should we copy the code and make a new week panel?

Just do it if need to

AdamsTsui commented:
Good job，Thanks!

FlareZh commented:
Any progress?

Sepush commented:
cc @07akioni  need code review

suica commented:
Hi, @07akioni ! i need your general comment on this implementation to make sure I am on the right track, so I can finish the rest todos :)

Sepush commented:
Thank you for your contribution. @suica 
would you mind add a week number row in the panel with gray color and a dot `.` like  `26.`
I think we can do it in another PR.

deiiviidas commented:
also need week days translate for example
near weekFormat, [weekDaysFormat ??]

weekDayLong =>[ 'Monday' , ....]
weekDayShort =>[ 'Mon' , ...]
